fem/libceed/ceed.cpp

```diff
@@ -258,14 +344,51 @@ void InitCeedBasisAndRestriction(const FiniteElementSpace &fes,
                                  Ceed ceed, CeedBasis *basis,
                                  CeedElemRestriction *restr)
 {
-   if (UsesTensorBasis(fes))
+   // Check for FES -> basis, restriction in hash tables
+   const Mesh *mesh = fes.GetMesh();
+   const FiniteElement *fe = fes.GetFE(0);
+   const int P = fe->GetDof();
+   const int Q = irm.GetNPoints();
+   const int nelem = mesh->GetNE();
+   CeedBasisKey basis_key(&fes, &irm, P, Q);
+   auto basis_itr = internal::ceed_basis_map.find(basis_key);
+   CeedRestrKey restr_key(&fes, P, nelem);
+   auto restr_itr = internal::ceed_restr_map.find(restr_key);
```

Comment:
These keys don't seem to uniquely define a basis and restriction and result into a bug.

Reply:
The key was ok (I think) but a stronger key doesn't hurt.

It was a good old 'double free' type issue.